reflex/ivars/number.py

```diff
@@ -1399,26 +1383,100 @@ class ToBooleanVarOperation(BooleanVar):
     """Base class for immutable boolean vars that are the result of a boolean operation."""
 
     _original_value: Var = dataclasses.field(
-        default_factory=lambda: LiteralBooleanVar(False)
+        default_factory=lambda: LiteralBooleanVar.create(False)
     )
 
-    def __init__(
-        self,
-        _original_value: Var,
+    @cached_property
+    def _cached_var_name(self) -> str:
+        """The name of the var.
+
+        Returns:
+            The name of the var.
+        """
+        return f"Boolean({str(self._original_value)})"
+
+    def __getattr__(self, name: str) -> Any:
+        """Get an attribute of the var.
+
+        Args:
+            name: The name of the attribute.
+
+        Returns:
+            The attribute value.
+        """
+        if name == "_var_name":
+            return self._cached_var_name
+        getattr(super(ToBooleanVarOperation, self), name)
+
+    @cached_property
+    def _cached_get_all_var_data(self) -> ImmutableVarData | None:
+        """Get all VarData associated with the Var.
+
+        Returns:
+            The VarData of the components and all of its children.
+        """
+        return ImmutableVarData.merge(
+            self._original_value._get_all_var_data(), self._var_data
+        )
+
+    def _get_all_var_data(self) -> ImmutableVarData | None:
+        return self._cached_get_all_var_data
+
+    def __hash__(self) -> int:
+        """Calculate the hash value of the object.
+
+        Returns:
+            int: The hash value of the object.
+        """
+        return hash((self.__class__.__name__, self._original_value))
+
+    def __post_init__(self):
+        """Post initialization."""
+        object.__delattr__(self, "_var_name")
+
+    @classmethod
+    def create(
+        cls,
+        value: Var,
         _var_data: VarData | None = None,
     ):
-        """Initialize the boolean var.
+        """Create the boolean var.
 
         Args:
-            _original_value: The original value.
+            value: The value of the var.
             _var_data: Additional hooks and imports associated with the Var.
+
+        Returns:
+            The boolean var.
         """
-        super(ToBooleanVarOperation, self).__init__(
+        return cls(
             _var_name="",
             _var_type=bool,
             _var_data=ImmutableVarData.merge(_var_data),
+            _original_value=value,
         )
-        object.__setattr__(self, "_original_value", _original_value)
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class TernaryOperator(ImmutableVar):
```

Comment:
i think this should be in `base.py` or a separate module. doesn't seem like it fits with the other `number.py` types because it fundamentally is not a number but the type if either `_if_true` or `_if_false`

Reply:
the reason i put it there in the first places because the condition is of `BooleanVar` type which is in the same file